src/App.tsx

```diff
@@ -1,10 +1,95 @@
+/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { addTodos, getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>('');
+  const [filter, setFilter] = useState<Filter>('all');
+  const [title, setTitle] = useState('');
+  const [addingTodoId, setAddingTodoId] = useState<number | null>(null);
+
+  const showError = (message: string) => {
+    setError(message);
+    setTimeout(() => setError(null), 3000);
+  };
+
+  const hideError = () => setError(null);
```

Comment:
```suggestion
  const hideError = () => setError('');
```
redundant to set null. initialState is empty string so clean error can be also empty string. And empty string can be one of fields in Error enum.